src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -483,7 +548,10 @@ FlowNode hitParallelStart() {
                 }
             }
         } else {
-            throw new IllegalStateException("Hit a BlockStartNode with multiple children, and no record of the start!");
+            if (isWalkingFromFinish()) {  //Incomplete single-branch parallels can do this
+                throw new IllegalStateException("Hit a BlockStartNode with no record of the start!");
+            }
+            return myCurrent.getParents().get(0); // No branches to explore
```

Comment:
Is it guaranteed that the list will be non `null` and contain at least one element?